src/caffe/layers/image_data_layer.cpp

```diff
@@ -37,10 +37,13 @@ void ImageDataLayer<Dtype>::DataLayerSetUp(const vector<Blob<Dtype>*>& bottom,
   const string& source = this->layer_param_.image_data_param().source();
   LOG(INFO) << "Opening file " << source;
   std::ifstream infile(source.c_str());
-  string filename;
+  std::string line;
```

Comment:
Why did we switch from `string` to `std::string` here? If we're going to `using` things (and we do `using std::string` in `common.hpp`), then I'd rather consistently drop the namespace.